packages/router/src/components/RouterProvider.tsx

```diff
@@ -27,7 +23,11 @@ function RouterContextProvider({
   router.update({
     ...router.options,
     ...rest,
+    // @ts-expect-error router options do not have context as property and I was unable to find any usage of it,
+    //  so is this spread required?
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     context: {
+      // @ts-expect-error see previous
       ...router.options.context,
     },
   })
```

Comment:
```suggestion
  router.update({
    ...router.options,
    ...rest,
  })
```

Reply:
Yeah, there might be things left while I was investigating Tanstack router. Good to me

Reply:
As per [comment below](https://github.com/tuono-labs/tuono/pull/147#issuecomment-2509196235) this will be handled in a separate task